lib/PipelineC/Tracing/Runner.cpp

```diff
@@ -0,0 +1,429 @@
+    if (Name == NullPointer) {
+      return;
+    }
+    Pointers[Name] = reinterpret_cast<uintptr_t>(Pointer);
+  }
+
+  template<typename T>
+  // clang-format off
+  requires std::is_pointer_v<T>
+  T getPointer(const llvm::StringRef Name, bool Invalidate) {
+    // clang-format on
+    if (Name == NullPointer)
+      return nullptr;
+
+    revng_check(Pointers.count(Name) != 0,
+                "Tried to retrieve a pointer from context that was not "
+                "previously stored");
+    T Result = (T) Pointers[Name];
+    if (Invalidate)
+      Pointers.erase(Name);
+
+    return Result;
+  }
+
+  // clang-format off
+private:
+
+  template<ConstexprString Name, typename ArgT, size_t I>
+  requires (anyOf<ArgT, char *, const char *>())
+  ArgStorage<ArgT> parseString(ArgumentRef Argument) {
+    // clang-format on
+    revng_check(Argument.isScalar(), "Argument is not scalar");
+    constexpr int LH = LengthHint<Name, I>;
+    ArgStorage<ArgT> Result;
+    if constexpr (LH > 0) {
+      AbortOnError(llvm::decodeBase64(Argument.getScalar(), Result));
+    } else {
+      Result.assign(Argument.getScalar().begin(), Argument.getScalar().end());
+    }
+    // Zero-terminate the string
+    Result.push_back(0);
+    return Result;
+  }
+
+  template<ConstexprString Name, IntegerType ArgT>
+  ArgStorage<ArgT> parseArgumentImpl(ArgumentRef Argument) {
+    return Argument.asInt<ArgT>();
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (anyOf<ArgT, char **, const char **>())
+  ArgStorage<ArgT> parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    return utils::toStringList(Argument);
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (std::is_pointer_v<ArgT> &&
+            isInteger<std::remove_pointer_t<ArgT>>())
+  ArgStorage<ArgT> parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    return Argument.toIntList<std::remove_pointer_t<ArgT>>();
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (std::is_pointer_v<ArgT> && isRPType<remove_constptr<ArgT>>())
+  ArgStorage<ArgT> parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    revng_check(Argument.isScalar(), "Argument is not scalar");
+    return getPointer<ArgT>(Argument.getScalar(), isDestroy<Name>());
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (isList<ArgT>() &&
+            isRPType<remove_constptr<remove_constptr<ArgT>>>())
+  ArgStorage<ArgT> parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    revng_check(Argument.isSequence(), "Argument is not a sequence");
+    using ElementT = std::remove_pointer_t<ArgT>;
+    std::vector<ElementT> Result;
+    for (auto &Elem : Argument.getSequence()) {
+      Result.push_back(getPointer<ElementT>(Elem, isDestroy<Name>()));
+    }
+
+    utils::poisonTail(Result);
+    return Result;
+  }
+
+public :
```

Comment:
```suggestion
public:
```

Reply:
See [the comment below](https://github.com/revng/revng/pull/193#discussion_r1160723068)